src/background/services/wallet/WalletService.ts

```diff
@@ -530,12 +550,30 @@ export class WalletService implements OnUnlock {
     tabId?: number,
     originalRequestMethod?: string,
   ): Promise<SigningResult> {
-    const wallet = await this.getWallet({ network, tabId });
+    const wallet = await this.getWallet({
+      network,
+      tabId,
+    });
 
     if (!wallet) {
       throw new Error('Wallet not found');
     }
 
+    if (isSolanaSigningRequest(tx)) {
+      if (wallet instanceof SolanaSigner) {
+        const signedTx = await wallet.signTx(
+          tx.data,
+          (await getProviderForNetwork(network)) as SolanaProvider,
+        );
+
+        return {
+          signedTx,
+        };
+      }
+
+      throw new Error('Unable to find a proper signer');
```

Comment:
Nit: We should throw early
```suggestion
      if (!(wallet instanceof SolanaSigner)) {
        throw new Error('Unable to find a proper signer');
      }
      const signedTx = await wallet.signTx(
        tx.data,
        (await getProviderForNetwork(network)) as SolanaProvider,
      );

      return {
        signedTx,
      };
```

Reply:
Fixed here: 8bb9887c0